02-oop-inheritance/script.js

```diff
@@ -0,0 +1,109 @@
+// 1 DONE- 4 DONE - 8 DONE
+function Movie(title, year, duration) {
+	this.inheritFrom = EventEmitter;
+  	this.inheritFrom();
+	this.title = title;
+	this.year = year;
+	this.duration = duration;
+	this.cast = new Array();
+	this.play = function() {
+		this.emit("play");
+	};
+	this.pause = function(){
+		this.emit("pause");
+	};
+	this.resume = function(){
+		this.emit("resume");
+	}
+	this.addCast = function(actors) {
+		this.cast = this.cast.concat(actors);
+	};
+	this.showCast = function() {
+		this.cast.forEach(function(actor) {
+			console.log(actor);
+		});
+	}
+}
+
+// 2 DONE
+let movie1 = new Movie("Notebook", 2004, 123);
+let movie2 = new Movie("Star Wars: Episode II - Attack of the Clones", 2002, 142);
+let movie3 = new Movie("SlumdogMillonaire", 2008, 120);
+
+// 3 DONE
+function EventEmitter() {
+	this.eventHandlers = [
```

Comment:
this list should be empty, and subclasses should populate it calling event emitter `on` method